### PR TITLE
Provide more context in the docs intro page

### LINKED ISSUE #22469: Make the docs landing page direct visitors in a more opinionated, descriptive way
## Applies To
[`docs/pages/index.mdx`](https://github.com/gravitational/teleport/blob/fe2ae5fac57260eb63f92d0b65b692a40e9fc54e/docs/pages/index.mdx)

## Details

After the first paragraph, the docs landing page directs users through different sections of the documentation, listing links for each section. There are two major downsides to this approach:

- **Too many ways to try out Teleport:** There are five links in the "Try out Teleport" section of the landing page, making it unclear how new users should begin the journey of learning Teleport. We should find a way to direct new users down a single path through the docs, ideally one that is easy to begin and demonstrates Teleport's value almost immediately.
- **No context around the different docs sections:** At the time I created this issue, there were 18 sidebar sections of the documentation. The only way to find out what these are for is to click on the sidebar, then visit the "Introduction" page for a section (or, if there's no "Introduction" page, doing your best based on the links in the sidebar).

What I propose:
- Structuring the landing page around _paragraphs_ rather than lists of links
- Including a single link for new users to try out Teleport
- For each of the docs sidebar sections, include a brief paragraph indicating what that section is about and linking to an introduction page. If there's no introduction page for a section, we can add one.

## Related Issues

https://github.com/gravitational/docs/issues/93

### PR DESCRIPTION
**Pull request by ptgott**
Closes #22469

The current docs landing page does not include enough context for new readers. After the general description of Teleport, the rest of the landing page is series of links, with little guidance for what a new user should do first (or what an intermediate user should do after completing the initial setup steps).

This change takes a more narrative approach to the landing page, briefly describing the content of each top-level docs section and putting it in context with the rest of the docs. This also helps users find the right top-level docs section for their needs, since otherwise the only way to do this is to visit each section.

This change is also more opinionated about which guide new users should begin with. It recommends the guide to getting started with Teleport Cloud, since this doesn't require setting up a Teleport cluster and helps a user register a resource quickly.

The landing page links to the introduction page of each top-level docs section. If a section does not have an introduction page, this change adds one.

### CODE REVIEW

File: docs/pages/architecture/introduction.mdx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Teleport Architecture Guides
+description: Get detailed information about how Teleport works
+---
+
+In this section, you will find detailed information about Teleport's internal
+architecture. Read these guides if you are interested in learning how Teleport
+works.
+
+- [Trusted Clusters](./trustedclusters.mdx)
```
Comment (zmb3) [inline docs/pages/architecture/introduction.mdx:10]: I'd suggest a slightly different oder:

- Authentication
- Proxy Service
- Teleport Nodes
- Session recording
- TLS routing
- Proxy peering
- Trusted clusters

Reason being:

- Trusted clusters are complicated, come with additional maintenance burden, and many of the use cases that would have required trusted clusters in the past can now be done with a more basic set up.
- This order introduces the "101" content first that is essential to understanding teleport, and moves advanced features like proxy peering after the basics.

File: docs/pages/index.mdx [hunk 2]
```diff
@@ -1,55 +1,131 @@
...
 
-Use Teleport to provide secure access to all of your infrastructure.
+Teleport is protocol aware and provides functionality that is unique to each
+protocol it supports. To enable access to a protocol, deploy the appropriate
+Teleport service and configure it to communicate with resources in your
+infrastructure. 
+
+Read about how to enable access to:
+
+- [Servers](./server-access/getting-started.mdx)
```
Comment (zmb3) [inline docs/pages/index.mdx:99]: Let's also order these in terms of maturity/adoption:

- Servers
- Kube Clusters
- Databases
- Applications
- Desktops

**Review by zmb3 — APPROVED** [review]

**Comment by ptgott** [comment]
@alexfornuto This should be ready for a copy review

File: docs/pages/management/introduction.mdx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+title: "Manage your Cluster"
+description: "Guides for performing day-two operations on your Teleport cluster."
+layout: tocless-doc
+---
+
+In this section, you can find guides for how to manage a Teleport cluster after
```
Comment (alexfornuto) [inline docs/pages/management/introduction.mdx:7]:
```suggestion
In this section, you can find guides on how to manage a Teleport cluster after
```

or 

```suggestion
In this section, you can find guides for on managing a Teleport cluster after
```

File: docs/pages/management/introduction.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+deploying it. 
+
+## Export audit events
+
+Teleport's Auth Service maintains an audit log that tracks events in your
+cluster, such as newly issued certificates. You can get insight into how users
```
Comment (alexfornuto) [inline docs/pages/management/introduction.mdx:13]: I suggest either adding more examples or removing the "such as" part.

File: docs/pages/management/introduction.mdx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+While Teleport is designed with security in mind, there are steps you can take
+to ensure that your cluster is as secure as possible. Read about these in the
+[Security](./security.mdx) section.
+
+## Admin guides
+
+This section describes how to perform routine administrative tasks in your
+Teleport cluster, such as upgrading the `teleport` binary or adding labels to
+Teleport resources. Read the [Admin Guides](./admin.mdx) section.
```
Comment (alexfornuto) [inline docs/pages/management/introduction.mdx:28]:
```suggestion
The [Admin Guides](./admin.mdx) section describes how to perform routine
administrative tasks in your Teleport cluster, such as upgrading the `teleport`
binary or adding labels to Teleport resources.
```

File: docs/pages/management/introduction.mdx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+## Operations
+
+The [Operations](./operations.mdx) section describes how to perform critical,
+infrequent tasks, such as achieving scalability, upgrading a cluster, and
+rotating Teleport's certificate authority.
```
Comment (alexfornuto) [inline docs/pages/management/introduction.mdx:34]:
```suggestion
The [Operations](./operations.mdx) section describes how to perform critical,
infrequent tasks such as achieving scalability, upgrading a cluster, and
rotating Teleport's certificate authority.
```

File: docs/pages/reference/introduction.mdx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  resources.
+- [Audit Events](./audit.mdx): A reference of Teleport's audit events.
+- [Helm Reference](./helm-reference.mdx): References for Teleport's Helm charts.
+- [Networking](./networking.mdx): Ports that Teleport services listen on, plus
+  other information about how Teleport services communicate.
+- [Authentication](./authentication.mdx): Teleport's authentication connectors.
+- [Signals](./signals.mdx): The signals you can use to control a Teleport
+  binary.
+- [Backends](./backends.mdx): Configuration options and required setup steps for
+  Teleport's Auth Service backends.
```
Comment (alexfornuto) [inline docs/pages/reference/introduction.mdx:23]:
```suggestion
- [Backends](./backends.mdx): Configuration options and required setup steps for
  Teleport's Auth Service storage backends.
```

**Review by alexfornuto — COMMENTED** [review]
I think it would be swell if we could apply a slug to the top-level nav item for these sections to point to these pages instead of an Introduction sub-page, but that's outside of the scope of this PR.

File: docs/pages/index.mdx [hunk 8]
```diff
@@ -1,55 +1,131 @@
...
+
+After trying out Teleport, you are ready to deploy a cluster to your
+infrastructure. Teleport has three editions, Teleport Enterprise, Teleport
+Enterprise Cloud, and Teleport Community Edition, and you can compare these in
+our [Choose an Edition](./choose-an-edition/introduction.mdx) section.
+
+<Notice type="tip">
+
+You can view information specific to an edition of Teleport by using the
+dropdown menu at the top of the page.
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/index.mdx:61]: Oops, fixed

File: docs/pages/index.mdx [hunk 9]
```diff
@@ -1,55 +1,131 @@
 ---
 title: Introduction to Teleport
 description: How to install and quickly get up and running with Teleport.
-h1: Introduction
 layout: tocless-doc
 videoBanner: ki-uVTSocGE
 ---
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/index.mdx:7]: There's before the first section in a `Notice` component—is there a more prominent way to show this?

**Review by klizhentas — APPROVED** [review]

**Review by alexfornuto — APPROVED** [review]

**Pull request merged**